test/unit/core/rendering.js

```diff
@@ -89,6 +89,32 @@ suite('Rendering', function() {
       assert.equal(fbo.width, 5);
       assert.equal(fbo.height, 15);
     });
+
+    test('should resize the dimensions of canvas based on max texture size', function() {
+      let glStub;
+      glStub = sinon.stub(p5.RendererGL.prototype, '_getParam');
+      const fakeMaxTextureSize = 100;
+      glStub.returns(fakeMaxTextureSize);
+      myp5.createCanvas(200, 200, myp5.WEBGL);
+      assert.equal(myp5.canvas.width, 100);
+      assert.equal(myp5.canvas.height, 100);
+
+      glStub.restore();
+    });
+
+    test('should resize the dimensions of canvas by resizeCanvas based on max texture size', function() {
+
+      let glStub;
+      glStub = sinon.stub(p5.RendererGL.prototype, '_getParam');
+      const fakeMaxTextureSize = 100;
+      glStub.returns(fakeMaxTextureSize);
+      myp5.createCanvas(10, 10, myp5.WEBGL);
+      myp5.resizeCanvas(200, 200);
+      assert.equal(myp5.canvas.width, 100);
+      assert.equal(myp5.canvas.height, 100);
```

Comment:
Do you think `myp5.width` and `myp5.height` should also be 100 here? I think currently they aren't because we set `this.width` and `this.height` before adjusting the sizes, but I'm not sure just yet what the expected result would be. I think I lean towards having those values also be 100 for consistency, what do you think?